src/cubicbez.rs

```diff
@@ -73,6 +76,224 @@ impl CubicBez {
+        let dt = 1.0 / n as f64;
+        let delta_2 = dt * dt;
+        let delta_3 = dt * delta_2;
+
+        std::iter::from_fn(move || {
+            // if storage exists, we use it exclusively
+            if let Some(storage) = storage.as_mut() {
+                return storage.pop();
+            }
+
+            // if storage does not exist, we are exclusively working down here.
+            if i >= n {
+                return None;
+            }
+
+            let t1 = i as f64 * dt;
+            let t1_2 = t1 * t1;
+            let a1 = a * delta_3;
+            let b1 = (3.0 * a * t1 + b) * delta_2;
+            let c1 = (2.0 * b * t1 + c + 3.0 * a * t1_2) * dt;
+            let d1 = a * t1 * t1_2 + b * t1_2 + c * t1 + d;
+            let result = CubicBez::from_parameters(a1, b1, c1, d1);
+            i += 1;
+            Some(result)
+        })
+    }
+
+    fn parameters(&self) -> (Vec2, Vec2, Vec2, Vec2) {
+        let c = (self.p1 - self.p0) * 3.0;
+        let b = (self.p2 - self.p1) * 3.0 - c;
+        let d = self.p0.to_vec2();
+        let a = self.p3.to_vec2() - d - c - b;
+        (a, b, c, d)
+    }
+
+    fn from_parameters(a: Vec2, b: Vec2, c: Vec2, d: Vec2) -> Self {
+        CubicBez::new(
+            d.to_point(),
+            d.to_point() + (c / 3.0),
+            d.to_point() + (c / 3.0) + (b + c) / 3.0,
+            d.to_point() + (c / 3.0) + (b + c) / 3.0 + a,
+        )
+    }
+
+    fn subdivide_3(&self) -> (CubicBez, CubicBez, CubicBez) {
+        let (p0, p1, p2, p3) = (
+            self.p0.to_vec2(),
+            self.p1.to_vec2(),
+            self.p2.to_vec2(),
+            self.p3.to_vec2(),
+        );
+        let mid1 = ((8.0 * p0 + 12.0 * p1 + 6.0 * p2 + p3) / 27.0).to_point();
+        let deriv1 = (p3 + 3.0 * p2 - 4.0 * p0) / 27.0;
+        let mid2 = ((p0 + 6.0 * p1 + 12.0 * p2 + 8.0 * p3) / 27.0).to_point();
+        let deriv2 = (4.0 * p3 - 3.0 * p1 - p0) / 27.0;
+        let left = CubicBez::new(
+            self.p0,
+            ((2.0 * p0 + p1) / 3.0).to_point(),
+            mid1 - deriv1,
+            mid1,
+        );
+        let mid = CubicBez::new(mid1, mid1 + deriv1, mid2 - deriv2, mid2);
+        let right = CubicBez::new(
+            mid2,
+            mid2 + deriv2,
+            ((p2 + 2.0 * p3) / 3.0).to_point(),
+            self.p3,
+        );
+        (left, mid, right)
+    }
+
+    /// Does this curve fit inside the given distance from the origin?
+    fn fit_inside(&self, distance: f64) -> bool {
+        if self.p2.to_vec2().hypot() <= distance && self.p1.to_vec2().hypot() <= distance {
+            return true;
+        }
+        let mid =
+            (self.p0.to_vec2() + 3.0 * (self.p1.to_vec2() + self.p2.to_vec2()) + self.p3.to_vec2())
+                * 0.125;
+        if mid.hypot() > distance {
+            return false;
+        }
+        // Split in two. Note that cu2qu here uses a 3/8 subdivision. I don't know why.
```

Comment:
Humm. Does it?
https://github.com/fonttools/fonttools/blob/48f68877a8ab5f868f57926b14eb4af1c811dfae/Lib/fontTools/cu2qu/cu2qu.py#L169-L174

looks like split at .5 to me.